rust/otap-dataflow/crates/otap/src/otap_exporter.rs

```diff
@@ -174,21 +194,29 @@ impl local::Exporter<OtapPdata> for OTAPExporter {
                             _ = arrow_metrics_client
                                 .arrow_metrics(request_stream)
                                 .await
-                                .map_err(|error| Error::ExporterError {
-                                    exporter: effect_handler.exporter_id(),
-                                    error: error.to_string(),
+                                .map_err(|e| {
+                                    self.pdata_metrics.metrics_failed.inc();
+                                    Error::ExporterError {
+                                        exporter: exporter_id.clone(),
+                                        error: e.to_string(),
+                                    }
                                 })?;
+                            self.pdata_metrics.metrics_exported.inc();
```

Comment:
This metric is tracking the number of OTAP batches. Is that what we want, or should it be the number of logs (e.g. number of rows in the `Logs` record batch)?

Reply:
Ideally, we are able to count (and limit) by at least four weights (requests, items, uncompressed bytes and compressed bytes). As a side-note, the "items" count is OTel's standard definition, it's a count of spans, logs records, metric data opints, and profiles.

Reply:
@albertlockett yea, I'm still a bit undecided about the pdata API, but I feel that what we have is a little too complicated and somewhat error-prone. It's going to be an interesting exercise to discuss.